src/modules/lorem/index.ts

```
@@ -18,20 +18,36 @@ export class Lorem {
    * Generates a word of a specified length.
    *
    * @param length length of the word that should be returned. Defaults to a random length.
+   * If a word with the specified length does not exist return a word with maximum possible length.
```

This does not fit my expectations regarding this parameter.

If the length of the word doesn't matter for me, then I wouldn't specify it.
The very fact, that I specify a specific length expresses intend that I actually want a word of that particular length.

If I just want a very long word, I dig into the data pick a suitable length and use that (Or maybe use a range parameter to narrow it down).

So this method should throw an error, if it cannot satisfy the requested data or should have an optional param/option that allows for a "closest fit". Not necessarily the longest. e.g. requesting a word with length 1 should not return `Rindfleischetikettierungsüberwachungsaufgabenübertragungsgesetz` (which is the name of an actual German law until 2013) according to https://de.babbel.com/de/magazine/das-laengste-wort.

---

This is not for general words, this is for `lorem`, which is the same for all languages (although it does change the alphabet for some, like jp).

I can change implementation to throw an error, but when I had a brief look, we don't do that in other places. For example, `faker.word.*` would return a word of any length, when length can not be satisfied: https://github.com/faker-js/faker/blob/main/src/modules/word/index.ts#L11

So if we want to stay consistent, that's fine for me. I can change it in any other place. The question is: should faker be strict or be as close to the expectation as possible?

---

I now looked a bit deeper into the implementation and tried to think about what your proposed implementation does.
And I'm more and more on @ST-DDT side (comment above here) that this current proposed PR is not what I, @ST-DDT or a user would expect.
We might need to discuss this in a team meeting, but I would currently reject your proposed solution, sorry 🙁

---

Yeah, let's discuss this at the meeting, as different functions behave differently. As @hankucz pointed out, `faker.word.*` would ignore length if it can not be satisfied. So if we want to throw an error, we would need to change the behavior of those other methods too, to stay consistent.